staging/src/k8s.io/apimachinery/pkg/util/intstr/intstr.go

```diff
@@ -155,7 +155,7 @@ func GetValueFromIntOrPercent(intOrPercent *IntOrString, total int, roundUp bool
 	if intOrPercent == nil {
 		return 0, errors.New("nil value for IntOrString")
 	}
-	value, isPercent, err := getIntOrPercentValue(intOrPercent)
+	value, isPercent, err := getIntOrPercentValueFromIntOrString(intOrPercent)
```

Comment:
`GetValueFromIntOrPercent` (the exported method that has out-of-tree callers) should still call getIntOrPercentValue and should be marked deprecated in favor of a corrected exported method.

Reply:
Created a new parallel function and added test cases extending the already existing ones to it. Also deprecated the old method.